mava/systems/jax/system_test.py

```diff
@@ -48,14 +47,18 @@ def dummy_str_plus_bool(self) -> None:
 
 
 # Mock builder
-class MockBuilder(Builder, MockCallbackHookMixin):
-    def __init__(self, components: List[Any]) -> None:
-        """Init for mock builder.
+class MockBuilder(MockCallbackHookMixin):
```

Comment:
isn't the point to test the actual mixin class so why are we creating a mock class?

Reply:
This was before we had the actual mixin class but I still think it is good to create mock versions to isolate the testing to just the system. But that said, we must write some integration tests for all the system processes.